EWL-6547: Resource page - mobile. tables not showing complete data

russom-woldezghi opened the pull request:
## Ticket(s)
- [EWL-6547: Resource page - mobile. tables not showing complete data](https://issues.ama-assn.org/browse/EWL-6547)
- [EWL-6548: Resource page - mobile. Tables grey/white not displaying correctly](https://issues.ama-assn.org/browse/EWL-6548)


## Description
Adds height to `td` table cells for mobile displays

Depends on https://github.com/AmericanMedicalAssociation/ama-d8/pull/1033

## To Test
- [ ] Checkout branch to `EWL-6547-mobile-showing-table-header`
- [ ] Run `gulp serve`
- [ ] Run `drush @one.local cr` in Drupal VM
- [ ] Reduce browser to mobile viewport
- [ ] Visit http://ama-one.local/about-ama/cpt-editorial-panel/cptr-editorial-panel-meeting-process-calendar#tables (check column title appear as data-title attributes - for ones with values)
- [ ] Visit http://ama-one.local/member-groups-sections/international-medical-graduates/imgs-meeting-agenda-important-dates#schedules
- [ ] Observe the table alternating row colors go from grey to white consistently
- [ ] Did you test in IE 11?

## Visual Regressions

N/A


## Relevant Screenshots/GIFs
### Before
![screen shot 2018-11-07 at 2 47 50 pm](https://user-images.githubusercontent.com/541745/48157419-89890000-e29d-11e8-9f57-900beda9765a.png)

### After
![screen shot 2018-11-07 at 2 47 05 pm](https://user-images.githubusercontent.com/541745/48157420-89890000-e29d-11e8-9c80-948f1d24320e.png)

## Remaining Tasks
Things relevant to here may be updating [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) with the latest naming changes or review of the relative [D8](https://github.com/AmericanMedicalAssociation/ama-d8) work.


## Additional Notes
Drupal PR this PR is dependent on: https://github.com/AmericanMedicalAssociation/ama-d8/pull/1033

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

Commit:
Increasing height for mobile viewport of td cells in tables

Commit:
Merge branch 'develop' into bugfix/EWL-6547-mobile-showing-table-header

russom-woldezghi commented:
There is a `source/assets/js/tables.js` in the style guide that is doing something similar, but not addressing the inclusion of `data-title` attributes to `td` of table elements. I opted to approach this fix from Drupal as that seems to address the issue. Let me know if there should be additional work around the the `table.js` file to avoid duplication.

damontgomery reviewed (commented):
Setting the height doesn't work for the example page.

![overlapping](https://user-images.githubusercontent.com/397902/48162387-8c3a2400-e2a2-11e8-8d88-5e5d2547c929.jpg)

Is there a dynamic solution that we can use that works for all lengths of titles?

Maybe a header above the content?

![stacked](https://user-images.githubusercontent.com/397902/48162427-ac69e300-e2a2-11e8-9adf-45d1f12b1533.jpg)

```
tbody tr td:before {
    font-weight: bold;
    text-transform: uppercase;
    content: attr(data-title);
    display: block;
}

tbody tr td {
    border: 1px solid #EDEDED;
    display: block;
    padding: 2px;
    position: relative;
    -webkit-hyphens: auto;
    -ms-hyphens: auto;
    hyphens: auto;
}
```

The testing steps in this PR and the D8 PR are very helpful. The comments are also very helpful. Thanks.

russom-woldezghi commented:
Hey @designerneil, for mobile display of tables, the two column layout in the styleguide: http://localhost:3000/?p=atoms-table might not be suitable for lengthy title/column header text in tables (see Dan's screenshot). Would Dan's suggest above need the styleguide, design assets such as Zeplin updated?

Commit:
Stacking table cells on mobile viewport

Commit:
Adding padding for mobile viewport

russom-woldezghi commented:
@damontgomery - I have added the suggestion you posted above, plus some additional padding so its not so close to the edge of the tables on mobile. 
![screen shot 2018-11-08 at 10 27 50 am](https://user-images.githubusercontent.com/541745/48208507-facdbf00-e340-11e8-9cb4-76e2d4405452.png)

For the schedule section: http://ama-one.local/member-groups-sections/international-medical-graduates/imgs-meeting-agenda-important-dates#schedules the content doesn't seem to match what is on the previous example (in the description/testing of the PR). This could be a content issue?

Commit:
Order of color for table rows

Commit:
Removing unused JS file

Commit:
Table twig file update with data-title attribute

russom-woldezghi commented:
@designerneil Can you take a look at this issue?  
![screen shot 2018-11-08 at 10 27 50 am](https://user-images.githubusercontent.com/541745/48282164-df3ce400-e426-11e8-8124-f23c3b1623a0.png)

I applied your changes from the other PR that is closed #525.

Brian-Clement commented on styleguide/source/assets/scss/01-atoms/_table.scss:
```diff
@@ -70,18 +70,20 @@ tbody {
       hyphens: auto;
 
+      &.odd,
+      &:nth-child(odd)  {
```
Changing this background from odd `tr` to odd `td` is creating the effect of column striping versus row striping.  Is this intentional?

russom-woldezghi replied:
I don't think the column striping in the desktop/tablet viewport is intentional. I can take a look. 

CC: @designerneil Do you have any thoughts on the background color for desktop/tablet viewport? 

The column stacking was decided to accommodate for lengthy title for mobile. It was a two-column before and text was getting cut off and overlapping. See below:

![image](https://user-images.githubusercontent.com/397902/48162387-8c3a2400-e2a2-11e8-8d88-5e5d2547c929.jpg)

Brian-Clement reviewed (commented):
@russom-woldezghi question on the striping.  The `background: $gray-7;` has moved from `tr` to `td`, which is creating striped columns instead of striped rows.  Not sure if this is intended or not, here are the effects on different viewport sizes:

# Desktop

## Before
![image](https://user-images.githubusercontent.com/4048700/48377240-e405d000-e692-11e8-8125-872506cf522f.png)


## After
![image](https://user-images.githubusercontent.com/4048700/48377261-f2ec8280-e692-11e8-9c5d-5df274308f3d.png)


# Tablet

## Before
![image](https://user-images.githubusercontent.com/4048700/48377289-0992d980-e693-11e8-8d3e-6b41a29d1303.png)


## After
![image](https://user-images.githubusercontent.com/4048700/48377305-13b4d800-e693-11e8-8462-54c4ca9a60dd.png)


# Mobile

## Before
![image](https://user-images.githubusercontent.com/4048700/48377321-216a5d80-e693-11e8-83b8-1ba55429854a.png)


## After
![image](https://user-images.githubusercontent.com/4048700/48377341-2a5b2f00-e693-11e8-9f3a-ab57fe4a2169.png)

Brian-Clement commented:
One other thing I'm seeing here @russom-woldezghi is the format of the table on mobile.  The column headers in the Zeplin are to the left of the cells (zpl.io/295JwNp), but in this PR they are appearing above every cell.

Commit:
More consistent stripping pattern between desktop, tablet and mobile stacking

russom-woldezghi commented:
@Brian-Clement Are you able to grant me access to the Zeplin account? I am not able to access the link. 

I have updated the stripping pattern to be a little more consistent across the desktop and tablet viewports. The update fixes the alternating rows, instead of columns for desktop and tablet. 

## Desktop
![screen shot 2018-11-13 at 10 59 38 am](https://user-images.githubusercontent.com/541745/48425804-8e353480-e733-11e8-97a4-c1def139286d.png)

## Tablet
![screen shot 2018-11-13 at 11 00 40 am](https://user-images.githubusercontent.com/541745/48425807-8ecdcb00-e733-11e8-9f4e-4ed43cf6d8fa.png)

## Mobile
The two gray background, circled in red (see below) are cases when stripping by odd or even can produce unintended background color order for mobile viewport in a stacking like display. We chose to stack the cells to accommodate for length titles, cell values. The two column layout caused the text to be overlapped and harder to read the content on mobile viewport. 
 
### Before
![48377341-2a5b2f00-e693-11e8-9f3a-ab57fe4a2169](https://user-images.githubusercontent.com/541745/48425471-e91a5c00-e732-11e8-9413-23add58685bc.png)

### After
I also added a bottom-border to highlight the background color consistency of cell background color on different viewports. 

![screen shot 2018-11-13 at 11 01 02 am](https://user-images.githubusercontent.com/541745/48425863-ae64f380-e733-11e8-9e58-fe0000f43ace.png)

damontgomery commented:
Since there are a few changes to tables that are overlapping, Neil is going to gather more information on the desired behavior and circle back.
